lightning/src/ln/channelmanager.rs

```diff
@@ -2090,12 +2090,24 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 				});
 			},
 			HTLCSource::PreviousHopData(hop_data) => {
+				let short_channel_id = hop_data.short_channel_id;
 				if let Err((their_node_id, err)) = match self.claim_funds_from_hop(&mut channel_state_lock, hop_data, payment_preimage) {
 					Ok(()) => Ok(()),
 					Err(None) => {
-						// TODO: There is probably a channel monitor somewhere that needs to
-						// learn the preimage as the channel already hit the chain and that's
-						// why it's missing.
+						let channel_state = &mut *channel_state_lock;
+						let chan_id = channel_state.short_to_id.get(&short_channel_id).unwrap().clone();
```

Comment:
Hmm, I dont think this would work - the channel in question likely no longer exists as we drop the Channel object once it hits the chain (and rely on ChannelMonitor). More likely need to be able to generate a ChannelMonitorUpdate with no channel knowledge using only the short_channel_id.